app/services/post_bookmarkable.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class PostBookmarkable < BaseBookmarkable
+  MODEL = Post
+  SERIALIZER = UserPostBookmarkSerializer
+
+  def initialize
+    super(MODEL, SERIALIZER)
+    @preload_associations = [{ topic: [:topic_users, :tags] }, :user]
```

Comment:
I wonder if this can be passed to the initializer instead of having to declare this as an instance variable.

Reply:
It could...I just thought it would be weird to have this as the only thing in the initializer. Might actually move this to a const like the others and pass to `super`

Reply:
I did this, it still looks a little funky but I need to be able to at least get the model from the const here:

https://github.com/discourse/discourse/blob/fdb440f08b0a19d2ad4a15ed78650281b75f9d90/app/models/bookmark.rb#L18-L21

Can you think of a nicer API for this model/serializer/preload associations trio?

Reply:
I think the base case can just define methods for it.

```
class BaseBookmarkable
  def model
    Post
  end

  def serializer
    PostBookmarkSerializer
  end

  def preload_associations
    [topic: { ... }]
  end
end
```

Reply:
Okay will do this, just thought it would be a little weird to use methods for this, but it also solves my problem in that I can make `model` a class method so I can access it via `bookmarkable_klass.model`

Reply:
Done